### ISSUE TITLE
Dokumentasjon - CSS

### ISSUE THREAD

**Issue by lha047** [Mattilsynet/designsystem#853]

**Comment by lha047** [comment]
For accessibility all animation and transitions should be in the media query `prefers-reduced-motion: no-preference`
```css
@media (prefers-reduced-motion: no-preference) {
  html {
    scroll-behavior: smooth;
  }
}
```

**Comment by lfbergee** [comment]
With a postcss plugin we can make our own media queries with custom props, that can make this a bit more user friendly with something like

`@media (--with-animation) {`

ref: https://github.com/csstools/postcss-plugins/tree/main/plugins/postcss-custom-media


I do think i prefer to flip this however. I known it can remove animations that could have stayed around, but it does make sure we never forget.

```
@media (prefers-reduced-motion: reduce) {
    *,
    *::after,
    *::before {
        animation-duration: 0ms !important;
        animation-delay: 0ms !important;
        transition: none !important;
    }
}
```

**Comment by lfbergee** [comment]
Perhaps it can be useful, with regards to tailwind, to also supply classes tailwind-style.

I know it looks scary for seasoned css-devs, but use with some care and it's fantastic for tailwinders.

```
<button class="btn btn-primary" ...
<a class="btn btn-ghost" ...
```

**Comment by lfbergee** [comment]
I assume most of our apps do not need js based animations. It might not be necessary for the design system to supply this, might be more of a reset thing that each application should supply itself.

And for sure, it's better to do it on a case by case basis if we are actually able to do that properly. However that makes it quite easy to make a mistake, so I prefer to just turn it off.

**Comment by petterkoding** [comment]
Yeah, 100%, Its totally fine in our case. I dont think we ever will use JS based animations or advanced animations at any point. The reset will do just fine and protect us from doing mistakes, like you said.
